kbfscrypto/encrypted_data.go

```diff
@@ -228,3 +228,59 @@ func DecryptTLFCryptKeys(
 
 	return oldKeys, nil
 }
+
+// EncryptedMerkleLeaf is an encrypted MerkleLeaf object.
+type EncryptedMerkleLeaf struct {
+	encryptedData
+}
+
+// MakeEncryptedMerkleLeaf constructs an EncryptedMerkleLeaf.
+func MakeEncryptedMerkleLeaf(
+	version EncryptionVer, data []byte, nonce *[24]byte) EncryptedMerkleLeaf {
+	return EncryptedMerkleLeaf{
+		encryptedData{
+			Version:       version,
+			EncryptedData: data,
+			Nonce:         nonce[:],
+		},
+	}
+}
+
+// PrepareMerkleLeaf verifies the correctness of the given leaf, and
+// returns its nonce.
+func PrepareMerkleLeaf(encryptedMerkleLeaf EncryptedMerkleLeaf) (
+	nonce [24]byte, err error) {
+	if encryptedMerkleLeaf.Version != EncryptionSecretbox {
```

Comment:
Should this be `<`, so it'll handle future versions?

Reply:
I don't follow how that would help handle future versions -- wouldn't `encryptedMerkleLeaf` be bigger in the case of a future version?  Since this version of the code supports exactly one version, it makes sense to me to use `!=`.  Later, if we introduce a new version, we'd have to change this code anyway to add a `switch` or something, right?

Reply:
Yeah that was the idea: a future version would have a higher `encryptedMerkleLeaf.Version`, and presumably the code was going to be compatible. If there is no such presumption, then you can leave it as `!=`.

Reply:
Ah, yeah I don't know how we can make this code compatible with a new version it doesn't know about yet.  Would be awesome if we could, but I don't see it happening.  Ideally we'd make this into a popup error so the user would know to upgrade, but that seems out of scope for now.